students/stefan_lund/Lesson_3/mailroom.py

```diff
@@ -0,0 +1,190 @@
+        if Name prompt for amount
+
+    """
+    # process tells if it's a start, thank you, amount or report
+    process = options[0]
+
+    if process == "start":
+        # start_options = ["start", "Send a Thank You", "Create a Report", "quit"]
+        start_form = "\nYour options are:\na.) {}\nb.) {}\nc.) {}".format(options[1], options[2], options[3])
+        print(start_form)
+        answer = input("\nEnter letter a, b or c according to your choice.: ")
+        return answer
+
+    elif process == "thank_you":
+        # thank_you_options = ["thank_you", "list", "Name", "quit"]
+        options_choice = options[1], options[2], options[3]
+        start_form = "\n  Your options are:\n  {}\n  {}\n  {}".format(*options_choice)
+        print(start_form)
+        input_prompt = "\nEnter option:   {}, {} of donor or {}: ".format(*options_choice)
+        answer = input(input_prompt)
+        return answer
+
+    elif process == "amount":
+        # amount_options = ["amount", "dollars or dollars.cents"]
+        print("\nEnter the amount.")
+        donation = input(options[1])
+        return donation
+
+    elif process == "report":
+        report(data)
+
+
+def thank_you(data):
+
+    thank_you_options = ["thank_you", "list", "Name", "quit"]
+
+    wrong_answer = True
+    question = user_input
+    while wrong_answer:
+        answer = question(thank_you_options, data)
+        accept = ["thank_you", "list", "quit"]
+        answer_list_or_quit = accept_user_input(answer, accept)
+        if answer_list_or_quit[1]:
+            answer = answer_list_or_quit[1]
+            if answer == "list":
+                report(data)
+            elif answer == "quit":
+                return False
+            else:
+                # assuming this must be a name and the program can continue
+                answer_list_or_quit = answer, False
+                wrong_answer = False
+
+    #  answer is a New Name or a Name already in data
+    name = answer
+    names = []
+    for old_name in data[::3]:
+        names.append(old_name)
+    if name not in names:
+        data.append(name)
+        data.append(0.00)
+        data.append(0)
+
+    name_ind = data.index(name)
+    wrong_answer = True
+    amount_options = ["amount", "dollars or dollars.cents: "]
+    accept = ["amount"]
+    while wrong_answer:
+        answer = user_input(amount_options, data)
+        if accept_user_input(answer, accept):
+            wrong_answer = False
+            amount = answer
+            if "." not in amount:
+                amount += '.00'
+            amount = round(float(amount), 2)
+            data[name_ind + 1] += amount
+            data[name_ind + 2] += 1
+
+    # print thank you note after accounting for the gift
+    thank_you_form = "\n\nDear {name},\n\nThank you for the very generous donation of ${amount}\n\nYours Truly,\n"
+    print(thank_you_form.format(name = name, amount = amount))

```

Comment:
same comment here, this function is a bit over-complicated - can you refactor to simplify? 


If you need some suggestions, I'd be happy to provide a code sample.